### DISCUSSION TITLE
Verwendung von DataLinq Query als Quelle für Dynamischen Inhalt?

### DISCUSSION THREAD

**Discussion by faulgurke** [e-netze/webgis-community#53] [Q&A]
Ist es möglich, die Antwort einer DataLinq Query  als Quelle für **dynamische Inhalte** zu verwenden?
Eine DataLinq Query liefert (Geo)JSON Werte. Die Idee wäre eine Query für dynamische Inhalte zu verwenden, weil man damit Marker aus Tabellen mit Koordinaten (ohne Geometrie) erstellen könnte.

Bislang ist mir das nicht gelungen, weil sich das Datenformat unterscheidet:
Gibt es eine Möglichkeit, die fehlenden Parameter zu ergänzen?
**GeoJSON**: 
{
"type": "FeatureCollection",
"name": "Zaehlstellen",
"crs": { "type": "name", "properties": { "name": "urn:ogc:def:crs:OGC:1.3:CRS84" } },
"features": [
{ "type": "Feature", "properties": { "Schluessel": "L58", "Name": "Mäderer Straße", "bei_km": 4.8959999, "ZST_Nummer": "9062_1",

**DataLinq Query**:
[
  {
    "ID": 324,
    "Str_Key": "L9",
    "Str_Bez": "Rucksteigstraße",
    "bei_km": 5.74,
    "AN_Typ": "automatisch",
    "Richtung1": "Möggers/Staatsgrenze",
    "Richtung2": "Hohenweiler/L1",
    "ZST_Nummer": "9093_1",

**Comment by gis-eni** [comment]
Eine DataLinq Quey lieft in der Regel in JSON zurück. Da Datalinq nicht vorrangig für Geo-Daten entworfen wurde, handelt es sich hier um ein allgemeines JSON zum beschreiben der Daten.

Es gibt aber die Möglichkeit, sich ein GeoJSON zurückgeben zu lassen. Dazu gibt man beim Aufruf der Query als "View" geojson an:

https://...../datalinq/select/endpoint-id@query-id@``geojson``

Diesen Link könntest du theoretisch auch als dynamischer Inhalt einfügen. Allerdings nur, wenn dein GeoJSON auch eine Geometrie hat. Das hast es nur, wenn dein DataLinq-JSON eine ``_location`` aufweist. Das reine DataLinq JSON (ohne ``geojson``) muss diese Eigenschaften für alle Objekte aufweisen:

``` javascript
...
"_location": {
      "Latitude": 47.18451581152855,
      "Longitude": 14.668355699796459,
      "BBox": null,
      "BBoxValid": false
    }
...
```
Erst dann hat dein GeoJSON auch eine Geometrie.
Bei SQL Abfragen mit Geo-Spalten bekommt das beispielsweise so:

``` SQL
SELECT 
[geom].STConvexHull().STCentroid().STAsText() as _location,
[geom].STSrid as _location_srid
FROM table
```

Wenn du nur x/y Spalten hast, musst du dir dein WKT Punkt Objekt irgendwie zusammenbauen. Performance!?
